scenarios/trycp_write_validated/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use holochain_types::prelude::ActionHash;
+use std::time::{Duration, Instant};
+use trycp_wind_tunnel_runner::embed_conductor_config;
+use trycp_wind_tunnel_runner::prelude::*;
+use validated_integrity::UpdateSampleEntryInput;
+
+embed_conductor_config!();
+
+#[derive(Debug, Default)]
+pub struct ScenarioValues {}
+
+impl UserValuesConstraint for ScenarioValues {}
+
+fn agent_setup(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+) -> HookResult {
+    connect_trycp_client(ctx)?;
+    reset_trycp_remote(ctx)?;
+
+    let client = ctx.get().trycp_client();
+    let agent_name = ctx.agent_name().to_string();
+
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            client
+                .configure_player(agent_name.clone(), conductor_config().to_string(), None)
+                .await?;
+
+            client.startup(agent_name.clone(), None).await?;
+
+            Ok(())
+        })?;
+
+    install_app(
+        ctx,
+        scenario_happ_path!("validated"),
+        &"validated".to_string(),
+    )?;
+    try_wait_for_min_peers(ctx, Duration::from_secs(120))?;
+
+    Ok(())
+}
+
+fn agent_behaviour(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+) -> HookResult {
+    let client = ctx.get().trycp_client();
+
+    let app_port = ctx.get().app_port();
+    let cell_id = ctx.get().cell_id();
+    let reporter = ctx.runner_context().reporter();
+
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            let start = Instant::now();
+
+            let action_hash: ActionHash = client
+                .call_zome(
+                    app_port,
+                    cell_id.clone(),
+                    "validated",
+                    "create_sample_entry",
+                    "this is a test entry value",
+                    Some(Duration::from_secs(80)),
+                )
+                .await
+                .map_err(|e| anyhow::anyhow!("call failure: {:?}", e))?
+                .decode()
+                .map_err(|e| anyhow::anyhow!("Decoding failure: {:?}", e))?;
+
+            reporter.add_custom(
+                ReportMetric::new("create_sample_entry_time")
+                    .with_field("value", start.elapsed().as_secs_f64()),
+            );
+
+            let start = Instant::now();
+
+            let _: ActionHash = client
+                .call_zome(
```

Comment:
The original wind tunnel runner had a `call_zome` common helper in `bindings/runner/common.rs` that does the decoding for you. I wonder if we should have one for the trycp version that grabs the information it needs from the context and does the call with encoding/decoding so that this become the 'raw' option and we can use the helper by default.

Doesn't have to be done on this PR and it's easy to swap one for the other if it's done later :)

Reply:
Good idea, this cleaned up the scenario a bunch: https://github.com/holochain/wind-tunnel/pull/77/commits/a7ad0d9ce0da34b331ddee5c59f9d6e621e75bbb

Reply:
Very nice, yeah if you can make the common functions do the nasty task spawn then the scenario ends up much cleaner